.github/workflows/js_tests.yml

```
@@ -12,27 +12,41 @@ permissions:
   contents: read
 
 jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        run: |
+          {
+            echo 'matrix<<EOF'
+            curl --silent --show-error --fail https://raw.githubusercontent.com/theforeman/foreman/${{ github.base_ref }}/.github/matrix.json
```

This means it won't work for forks. I'd prefer to use `actions/checkout` to get the real file.

---

Or you can use the correct namespace and repo name from variables.
```suggestion
            curl --silent --show-error --fail https://raw.githubusercontent.com/${{ github.repository }}/${{ github.base_ref }}/.github/matrix.json
```

---

eh? `base_ref` is the PR target, which I just assume to be `theforeman/foreman`.

or do you mean PRs against forks? :D

---

Hmm no. What I mean is that you should be able to create a PR that changes Ruby/Node and CI should respect that.

So what I'd expect is that you have the equivalent of `actions/checkout` and take `.github/matrix.json` from that.

---

Mhh, OK.

---

But you're right that `base_ref` is not right. I think it should be `ref` or `ref_name`.